src/desktop.vala

```diff
@@ -94,22 +94,19 @@ class Desktop: Profile
     var action_name = @"desktop-device-$(id)-connected";
 
     var item = new MenuItem (_("Connection"), @"indicator.$action_name");
-    item.set_attribute ("x-canonical-type",
-                        "s", "com.canonical.indicator.switch");
+    item.set_attribute ("x-ayatana-type",
+                        "s", "org.ayatana.indicator.switch");
 
     // if this doesn't already have an action, create one
     if (!connect_actions.contains (id))
       {
         debug (@"creating action for $action_name");
         var a = new SimpleAction.stateful (action_name,
-                                           null,
+                                           VariantType.BOOLEAN,
                                            new Variant.boolean (device.is_connected));
 
-        a.activate.connect (()
-          => a.set_state (new Variant.boolean (!a.get_state().get_boolean())));
-
-        a.notify["state"].connect (()
```

Comment:
Why did this get removed?

Reply:
Why did a.activate.connect get replaced with a.notify["state"].connect functions and a.notify["state"].connect gets completely removed

Reply:
I assume a.set_state set the state then a.notify["state"] will pick that up, but we need to set state dont we?

Reply:
This old approach sometimes fails to get the right state after activation. There is now a much more elegant and client-friendly way for IDO:

After activation (when the state has changed), the owner of the action receives the "activate" signal with the new state as a parameter. No need to subscribe to the state change or request a state change manually any more.